File: 프로그래머스/lv0/120890. 가까운 수/가까운 수.py
def solution(array, n):
    answer = 1e9
    d = 1e9
    for i in array:
        if abs(n - i) <= d:
            if abs(n-i) == d and answer < i:
                continue
            answer = i
            d = abs(n-i)
    return answer

File: 프로그래머스/0/120890. 가까운 수/가까운 수.java
import java.util.*;

class Solution {
    public int solution(int[] array, int n) {
        Arrays.sort(array);
        int answer = array[0];

        for (int i = 1; i < array.length; i++) {
            if (Math.abs(array[i] - n) < Math.abs(answer - n)) {
                answer = array[i];
            }
        }

        return answer;
    }
}